hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamerMetrics.java

```diff
@@ -67,12 +73,41 @@ String getMetricsName(String action, String metric) {
 
   public void updateDeltaStreamerMetrics(long durationInNs, long hiveSyncNs) {
     if (config.isMetricsOn()) {
-      Metrics.registerGauge(getMetricsName("deltastreamer", "duration"), getDurationInMs(durationInNs));
-      Metrics.registerGauge(getMetricsName("deltastreamer", "hiveSyncDuration"), getDurationInMs(hiveSyncNs));
+      registerOrUpdateMetrics(getMetricsName("deltastreamer", "duration"), getDurationInMs(durationInNs));
+      registerOrUpdateMetrics(getMetricsName("deltastreamer", "hiveSyncDuration"), getDurationInMs(hiveSyncNs));
     }
   }
 
   public long getDurationInMs(long ctxDuration) {
     return ctxDuration / 1000000;
   }
+  
+  public void registerOrUpdateMetrics(String metricsName, Object value) {
+    if (metrics.containsKey(metricsName)) {
+      metrics.get(metricsName).set(value);
```

Comment:
\after updated, the updated value would not be sent as well?

Reply:
Update value will be sent by MetricsGraphiteReporter, MetricsGraphiteReporter.start() will start GraphiteReporter and will report metric periodically.